vllm/engine/metrics.py

```diff
@@ -24,6 +24,9 @@
 gauge_cpu_cache_usage = Gauge(
     "vllm:cpu_cache_usage_perc",
     "CPU KV-cache usage. 1 means 100 percent usage.")
+gauge_max_step_time_since_log = Gauge(
+    "vllm:max_step_time_since_log_s",
```

Comment:
instead of max. I think we should record a running average with old * 0.9 + new * 0.1. 
Or just report how long the last iteration took or both. This max since log is kind or arbitrary based on the log interval. But prometheus might be scraping stats using different interval.

Reply:
So I would name it vllm:avg_gen_time_ms, vllm:last_gen_time_ms
Also we should make sure we do this for generation and not prefill

Reply:
I have created stats:
vllm:avg_gen_time_ms, vllm:max_gen_time_ms

I have split prefills into:
vllm:avg_prompt_time_ms, vllm:max_prompt_time_ms